.github/workflows/continuous-integration.yml

```
@@ -0,0 +1,188 @@
+name: "Continuous Integration"
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+  push:
+    branches:
+      - '[0-9]+.[0-9]+.x'
+      - 'refs/pull/*'
+    paths-ignore:
+      - 'mkdocs.yml'
+      - '**.md'
+
+jobs:
+  lint-composer-json:
+    name: Lint composer.json
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: json-syntax-check
+        run: jq empty composer.json
+
+  supported-checks-matrix:
+    name: Determine which checks to run
+    runs-on: ubuntu-latest
+    needs: [lint-composer-json]
+    outputs:
+      check: ${{ steps.supported-checks-matrix.outputs.check }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: supported-checks-matrix
+        run: |
+          # Update the tools def to add or exclude checks
+          CHECKS=$(cat composer.json | jq -c 'def tools:["benchmark","cs-check","static-analysis","test"] ; def intersect(x;y): (x + y | sort) as $sorted | reduce range(1; $sorted|length) as $i ([]; if $sorted[$i] == $sorted[$i-1] then . + [$sorted[$i]] else . end) ; .scripts | keys | intersect(.;tools)')
```

```suggestion
          CHECKS=$(jq -c 'def tools:["benchmark","cs-check","static-analysis","test"] ; def intersect(x;y): (x + y | sort) as $sorted | reduce range(1; $sorted|length) as $i ([]; if $sorted[$i] == $sorted[$i-1] then . + [$sorted[$i]] else . end) ; .scripts | keys | intersect(.;tools)' < cat composer.json)
```

also, perhaps some brief (as a comment in there) what that snippet does?